Core/include/Acts/Propagator/StepperConcept.hpp

```diff
@@ -9,215 +9,124 @@
-        // static_assert(const_component_proxy_exists, "!const_component_proxy_exists");
-
-        METHOD_TRAIT(number_components_t, numberComponents);
-        constexpr static bool number_components_exists = has_method<const S, std::size_t, number_components_t, const state&>;
-        // static_assert(number_components_exists, "!num_components_exists");
-
-        // TODO We cannot check addComponents since it is a template member function
-
-        METHOD_TRAIT(clear_components_t, clearComponents);
-        constexpr static bool clear_components_exists = has_method<const S, void, clear_components_t, state&>;
-        // static_assert(clear_components_exists, "!clear_components_exists");
-
-        METHOD_TRAIT(remove_missed_components_t, removeMissedComponents);
-        constexpr static bool remove_missed_components_exists = has_method<const S, void, remove_missed_components_t, state&>;
-        // static_assert(remove_missed_components_exists, "!remove_missed_components_exists");
-
-        constexpr static bool value = require<common_stepper_concept_fullfilled,
-                                              component_proxy_exists,
-                                              const_component_proxy_exists,
-                                              number_components_exists,
-                                              clear_components_exists,
-                                              remove_missed_components_exists>;
-        };
-// clang-format on
-
-}  // namespace Concepts::Stepper
-
-template <typename stepper, typename state = typename stepper::State>
-constexpr bool StepperConcept =
-    Acts::Concepts ::Stepper::SingleStepperConcept<stepper, state>::value ||
-    Acts::Concepts ::Stepper::MultiStepperConcept<stepper, state>::value;
-template <typename stepper>
-constexpr bool StepperStateConcept =
-    Acts::Concepts ::Stepper::StepperStateConcept<stepper> ||
-    Acts::Concepts ::Stepper::MultiStepperStateConcept<stepper>;
-}  // namespace Acts
+    };
+  };
+
+  requires requires(const Stepper& s, const State& t) {
+    { s.position(t) } -> std::same_as<Vector3>;
+    { s.direction(t) } -> std::same_as<Vector3>;
+    { s.qOverP(t) } -> std::same_as<double>;
+    { s.absoluteMomentum(t) } -> std::same_as<double>;
+    { s.momentum(t) } -> std::same_as<Vector3>;
+    { s.charge(t) } -> std::same_as<double>;
+    { s.time(t) } -> std::same_as<double>;
+    { s.outputStepSize(t) } -> std::same_as<std::string>;
+
+    requires requires(const ConstrainedStep::Type st) {
+      { s.getStepSize(t, st) } -> std::same_as<double>;
+    };
+  };
+};
+
+/// @brief Concept that is satisfied by single-steppers.
+template <typename Stepper, typename State = typename Stepper::State>
+concept SingleStepper = CommonStepper<Stepper, State> &&
+    requires(const Stepper& s, State& t) {
+  requires requires(const FreeVector& fv, const BoundVector& bv,
```

Comment:
What I've done in other places is just putting all variables in the top `requires` clause, and avoiding nested `requires` blocks. Ultimately, it doesn't matter though.